File: expresslk-services/ExpressLK-core/src/main/java/lk/express/rule/bean/Rule.java
package lk.express.rule.bean;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.xml.bind.annotation.XmlRootElement;
import javax.xml.bind.annotation.XmlType;

import lk.express.bean.HasValidPeriod;
import lk.express.rule.ApplicationType;

import org.hibernate.annotations.Cascade;
import org.hibernate.annotations.CascadeType;

@Entity
@Table(name = "rule")
@Inheritance(strategy = InheritanceType.JOINED)
@XmlType(name = "Rule", namespace = "http://bean.rule.express.lk")
@XmlRootElement
public abstract class Rule extends lk.express.bean.Entity implements HasValidPeriod {

	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue
	private Integer id;
	@Column(name = "name")
	private String name;
	@Column(name = "description", columnDefinition = "text")
	private String description;
	@Column(name = "salience")
	private float salience;
	@Column(name = "scheme")
	private String scheme;
	@ManyToOne(cascade = javax.persistence.CascadeType.PERSIST)
	@Cascade(CascadeType.SAVE_UPDATE)
	@JoinColumn(name = "rule_condition")
	private RuleCondition condition;
	@Column(name = "amount")
	private float amount;
	@Column(name = "application_type")
	@Enumerated(EnumType.STRING)
	private ApplicationType applicationType;
	@Temporal(TemporalType.TIMESTAMP)
	@Column(name = "start_time")
	private Date startTime;
	@Temporal(TemporalType.TIMESTAMP)
	@Column(name = "end_time")
	private Date endTime;

	@Override
	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public float getSalience() {
		return salience;
	}

	public void setSalience(float salience) {
		this.salience = salience;
	}

	public String getScheme() {
		return scheme;
	}

	public void setScheme(String scheme) {
		this.scheme = scheme;
	}

	public RuleCondition getCondition() {
		return condition;
	}

	public void setCondition(RuleCondition condition) {
		this.condition = condition;
	}

	public float getAmount() {
		return amount;
	}

	public void setAmount(float amount) {
		this.amount = amount;
	}

	public ApplicationType getApplicationType() {
		return applicationType;
	}

	public void setApplicationType(ApplicationType applicationType) {
		this.applicationType = applicationType;
	}

	@Override
	public Date getStartTime() {
		return startTime;
	}

	public void setStartTime(Date startTime) {
		this.startTime = startTime;
	}

	@Override
	public Date getEndTime() {
		return endTime;
	}

	public void setEndTime(Date endTime) {
		this.endTime = endTime;
	}
}

File: expresslk-webportal/module/Api/src/Api/Client/Soap/Core/Rule.php
<?php

namespace Api\Client\Soap\Core;

include_once('Entity.php');

class Rule extends Entity
{

    /**
     * @var float $amount
     * @access public
     */
    public $amount = null;

    /**
     * @var ApplicationType $applicationType
     * @access public
     */
    public $applicationType = null;

    /**
     * @var RuleCondition $condition
     * @access public
     */
    public $condition = null;

    /**
     * @var string $description
     * @access public
     */
    public $description = null;

    /**
     * @var dateTime $endTime
     * @access public
     */
    public $endTime = null;

    /**
     * @var int $id
     * @access public
     */
    public $id = null;

    /**
     * @var string $name
     * @access public
     */
    public $name = null;

    /**
     * @var float $salience
     * @access public
     */
    public $salience = null;

    /**
     * @var string $scheme
     * @access public
     */
    public $scheme = null;

    /**
     * @var dateTime $startTime
     * @access public
     */
    public $startTime = null;

    /**
     * @param float $amount
     * @param ApplicationType $applicationType
     * @param RuleCondition $condition
     * @param string $description
     * @param dateTime $endTime
     * @param int $id
     * @param string $name
     * @param float $salience
     * @param string $scheme
     * @param dateTime $startTime
     * @access public
     */
    public function __construct($amount, $applicationType, $condition, $description, $endTime, $id, $name, $salience, $scheme, $startTime)
    {
      $this->amount = $amount;
      $this->applicationType = $applicationType;
      $this->condition = $condition;
      $this->description = $description;
      $this->endTime = $endTime;
      $this->id = $id;
      $this->name = $name;
      $this->salience = $salience;
      $this->scheme = $scheme;
      $this->startTime = $startTime;
    }

}
